hackathon_site/hardware/serializers.py

```diff
@@ -46,9 +50,91 @@ def get_unique_hardware_count(obj: Category):
         return Hardware.objects.filter(categories__id=obj.id).count()
 
 
-class OrderSerializer(serializers.ModelSerializer):
+class OrderListSerializer(serializers.ModelSerializer):
     hardware_set = HardwareSerializer(many=True, read_only=True)
 
     class Meta:
         model = Order
         fields = ("id", "hardware_set", "team", "status", "created_at", "updated_at")
+
+
+class OrderCreateSerializer(serializers.Serializer):
+    class OrderCreateHardwareSerializer(serializers.Serializer):
+        id = serializers.PrimaryKeyRelatedField(
+            queryset=Hardware.objects.all(), many=False, required=True
+        )
+        quantity = serializers.IntegerField(required=True)
+
+    team_id = serializers.PrimaryKeyRelatedField(
+        queryset=TeamEvent.objects.all(), many=False, required=True
+    )
+    hardware = OrderCreateHardwareSerializer(many=True, required=True)
```

Comment:
we could change this to a dictionary instead of a nested serializer
nested serializer:
- 🟢dont need to validate hardware pk ourselves
- 🔴there can be multiple entries for the same hardware. additional logic is needed to merge them; or we can not merge them and process them regardless if you are okay with that)
dictionary:
- 🟢all entries have unique keys, wont have multiple entires for the same hardware
- 🔴need to validate the pk's ourselves in the serializer

Reply:
- In the jira ticket, it is a dictionary mapping hardware id to quantity
- Even if there are multiple hardware entries with the current list of dicts, that shouldn't be a problem - since we have to add N hardware objects to the order anyway, having hardware appear twice with N and M quantities just means we add N + M. Though I'm guessing that makes validation more annoying, hence why you use the Counter
- All entries having unique keys - technically, json doesn't enforce this... I'm not sure what django would actually do if you posted duplicate keys, probably just keep the last one
- I like not having to validate things ourselves :)

Tldr - I defer to you. Either list of dicts, or a straight dict is fine with me.

Reply:
@albogdan thoughts?